doc/rfc008-managing-apalache.md

```diff
@@ -0,0 +1,343 @@
+
+Cons:
+
+- Only pushes the boundary: e.g., this would assume that npm, coursier, or the
+  JRE are installed on the local system. `vscode-metals` includes a
+  `coursier` binary in its distribution to counter this.
+- Spawning a package manager increases complexity by running a process whose
+  behavior is outside of our control.
+
+#### 3.2.5 Apalache as cloud-hosted SaaS
+
+Quint uses Apalache via a public SaaS API, no local installation or management
+is necessary.
+
+Pros:
+
+- No installation or server management issues.
+- Versioning and backwards-compatiblity could be exposed via separate endpoints.
+
+Cons:
+
+- Relies on internet connection.
+- apalache-cloud is currently not exposed via the Shai gRPC interface.
+- Recurring cost for cloud hosting, likely requires billing of external users.
+- May deter external users for privacy concerns.
+
+### 3.3 Version dependencies between Quint and Apalache
+
+We want to maintain compatiblity between Quint and Apalache, therefore we need
+some mode of linking compatible versions of both tools. Both tools follow
+[semantic versioning][].
+
+The crux here is that both an upgrade and a downgrade of Apalache, relative to a
+specific Quint release, may maintain or break compatibility.
+
+Given that Quint calls Apalache, but not the other way around, it makes sense to
+maintain a list of compatible Apalache versions for a given Quint release.
+Broadly speaking, there's two options for maintaining such a list:
+
+1. Informally, as part of the release notes, or other documentation.
+2. Formally, inside the Quint codebase.
+
+Moreover, there's multiple options on how to specify compatible versions:
+
+#### 3.3.1 Use arbitrary Apalache release
+
+We don't expect a specific Apalache release from Quint.
+
+Pros:
+
+- No need to manage or communicate version dependencies.
+
+Cons:
+
+- Things break at runtime with incompatible versions, possibly in weird places
+  and with strange error messages or uncaught exceptions.
+
+#### 3.3.2 Use latest Apalache release
+
+Regardless of the Quint release, we expect the latest version of Apalache to be
+installed. We could use the Github REST API to check the latest Apalache release
+and abort if the installed version differs.
+
+Pros:
+
+- Easy management, no need to pin specific versions from Quint.
+
+Cons:
+
+- Prone to break if users have an outdated version of Quint.
+  - We could address this by also checking the latest release of Quint, and
+    alerting users to upgrade.
+- Prone to break if users downgrade their installation of Quint.
+
+#### 3.3.3 Pin a specific Apalache version
+
+Each Quint release pins a specific version of Apalache.
+
+Pros:
+
+- Enables user to downgrade Quint, while maintaining compatibility with Apalache.
+
+Cons:
+
+- Users don't receive Apalache bugfix releases, unless we cut another Quint
+  release that pins the updated dependency.
+
+#### 3.3.4 Pin a minor release branch
```

Comment:
This sounds like a good enough approach to me! Also, I think that at some point in the future, when apalache, quint and their integration are all more stable, we can evolve this into "Pin a major release branch" and not have to worry about the "Cons" point more than a couple of times per year :smile:

Reply:
I also like 3.3.4 the most. It would let us go ahead with the Quint and Apalache development while maintaining relatively stable versions of both.